hs/src/Reach/Connector/ETH_Solidity.hs

```diff
@@ -679,6 +680,11 @@ solExpr sp = \case
   DLE_Transfer _ who amt mtok ->
     spa $ solTransfer who amt mtok
   DLE_TokenInit {} -> return emptyDoc
+  DLE_CanReceive _ _ _ -> do
+    --addr' <- solArg addr
+    --tok' <- solArg tok
+    -- TODO - this is not really correct, because this is throwing away the arguments, which could potentially be effectful.  Also, this is not being generated anyway, so something is wrong before here, keeping me from getting here.
```

Comment:
No, they cannot be effectful. Reach's AST is like ANF, so its arguments are always simple and if they were effectful to generate, that would be annotate on their let